decidim-core/app/forms/decidim/registration_form.rb

```diff
@@ -41,6 +44,10 @@ def email_unique_in_organization
       errors.add :email, :taken if User.where(email: email, organization: current_organization).first.present?
     end
 
+    def nickname_unique_in_organization
+      errors.add :nickname, :taken if User.where(nickname: nickname, organization: current_organization).first.present?
+    end
```

Comment:
We have a lot of "duplicated validations" between the registration from and the `User` model. I just kept repeating them but maybe we can improve things in this regard in the future?

Reply:
I think they have different purposes. `models` deal with *data integrity* and forms with *data normalization & user input validation*. I think it's natural that some validations (like presence) are duplicated, but others (length, format) are form-specific. Does this make sense?

Reply:
When I checked some validations felt off at the model level, but I didn't want to explore deeper. But yeah, you're right.